src/mtev_http.c

```diff
@@ -981,6 +1008,13 @@ mtev_http_complete_request(mtev_http_session_ctx *ctx, int mask) {
     if(len <= 0) goto full_error;
     if(len > 0) in->size += len;
     rv = mtev_http_request_finalize(&ctx->req, &err);
+    /* walk the bchain and set the compression */
+    struct bchain *x = ctx->req.first_input;
+    while (x) {
+      x->compression = request_compression_type(&ctx->req);
+      ctx->req.content_length_read += x->size;
```

Comment:
This is handled by the fix for `rv == mtev_true`.  i.e. don't walk the input chain until the request is finalized.  At this point the only thing left in the bchain is request payload if any of the request payload has been read.